plugins/woocommerce/src/Internal/RestApi/Routes/V4/Settings/Tax/Controller.php

```diff
@@ -0,0 +1,417 @@
+<?php
+/**
+ * REST API Tax Settings Controller
+ *
+ * Handles requests to the /settings/tax endpoints.
+ *
+ * @package WooCommerce\RestApi
+ */
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Internal\RestApi\Routes\V4\Settings\Tax;
+
+use WP_Error;
+use Automattic\WooCommerce\Internal\RestApi\Routes\V4\AbstractController;
+use Automattic\WooCommerce\Internal\RestApi\Routes\V4\Settings\Tax\Schema\TaxSettingsSchema;
+use WP_REST_Server;
+use WP_REST_Request;
+use WP_REST_Response;
+use WC_Settings_Tax;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API Tax Settings Controller Class.
+ */
+class Controller extends AbstractController {
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'settings/tax';
+
+	/**
+	 * WC_Settings_Tax instance.
+	 *
+	 * @var WC_Settings_Tax
+	 */
+	protected $settings_tax_instance;
+
+	/**
+	 * Schema instance.
+	 *
+	 * @var TaxSettingsSchema
+	 */
+	protected $schema;
+
+	/**
+	 * Initialize the controller.
+	 *
+	 * @param TaxSettingsSchema $schema Schema class.
+	 * @internal
+	 */
+	final public function init( TaxSettingsSchema $schema ) {
+		$this->schema = $schema;
+	}
+
+	/**
+	 * Get the WC_Settings_Tax instance.
+	 *
+	 * @return WC_Settings_Tax
+	 */
+	private function get_settings_tax_instance() {
+		if ( is_null( $this->settings_tax_instance ) ) {
+			$this->settings_tax_instance = new WC_Settings_Tax();
+		}
+		return $this->settings_tax_instance;
+	}
```

Comment:
I think we can use the dependency injection here for the tax settings (like it is done for the schema). More info here: https://wooengineering.wordpress.com/2025/09/22/developing-rest-api-v4-routes/#1-dependency-injection

```suggestion
	final public function init( TaxSettingsSchema $schema, WC_Settings_Tax $settings_tax_instance) {
		$this->schema = $schema;
		$this->settings_tax_instance = $settings_tax_instance;
	}
```

Reply:
If I'm not wrong, that only works for classes in the same namespace. That's why the other settings controllers are doing it manually.